flink-connectors/flink-connector-files/src/main/java/org/apache/flink/connector/file/table/FileSystemOutputFormat.java

```diff
@@ -286,10 +302,21 @@ public Builder<T> setIsToLocal(boolean isToLocal) {
         }
 
         public Builder<T> setPath(Path path) {
+            return this.setStagingPath(toStagingPath(path));
+        }
```

Comment:
Nit: to avoid arch rule being violated
```suggestion
            this.path = toStagingPath(path);
            return this;
        }
```